lib/plugin.js

```diff
@@ -1,76 +1,100 @@
 'use strict'
 
 const fastq = require('fastq')
-const EE = require('events').EventEmitter
+const { EventEmitter } = require('events')
 const inherits = require('util').inherits
-const { debug } = require('./lib/debug')
-const { loadPlugin } = require('./lib/load-plugin')
-const { createPromise } = require('./lib/create-promise')
-const { AVV_ERR_READY_TIMEOUT } = require('./lib/errors')
-const { getPluginName } = require('./lib/get-plugin-name')
-const { isPromiseLike } = require('./lib/is-promise-like')
+const { debug } = require('./debug')
+const { loadPlugin } = require('./load-plugin')
+const { createPromise } = require('./create-promise')
+const { AVV_ERR_READY_TIMEOUT } = require('./errors')
+const { getPluginName } = require('./get-plugin-name')
+const { isPromiseLike } = require('./is-promise-like')
 
+/**
+ * @param {*} parent
+ * @param {*} func
+ * @param {*} options
+ * @param {boolean} isAfter
+ * @param {number} [timeout]
+ */
 function Plugin (parent, func, options, isAfter, timeout) {
-  this.started = false
-  this.func = func
-  this.opts = options
-  this.onFinish = null
   this.parent = parent
-  this.timeout = timeout === undefined ? parent._timeout : timeout
-  this.name = getPluginName(func, options)
+  this.func = func
+  this.options = options
+  /**
+   * @type {boolean}
+   */
   this.isAfter = isAfter
-  this.q = fastq(parent, loadPluginNextTick, 1)
-  this.q.pause()
+  /**
+   * @type {number}
+   */
+  this.timeout = timeout
+
+  /**
+   * @type {boolean}
+   */
+  this.started = false
+  /**
+   * @type {string}
+   */
+  this.name = getPluginName(func, options)
+
+  /**
+   * @type {fastq.queue}
+   */
+  this.queue = fastq(parent, loadPluginNextTick, 1)
```

Comment:
the only reason, we have to pass parent to the Plugin-constructor is that we have to instantiate a fastq-instance. 

Any idea how we could decouple it completely?

Reply:
I don't think it's possible to decouple them

Reply:
I think i successfully decoupled them. Now I pass a fastq queue as first param instead of the avvio instance.